rrfs-test/scripts/link_mpasjedi_expr.sh

```diff
@@ -45,7 +45,7 @@ cp ${RDASApp}/rrfs-test/ush/mpasjedi_spread.py .
 mkdir -p data
 cd data
 mkdir -p bumploc bkg obs ens ref
-ln -snf ${RDASApp}/fix/bumploc/${BUMPLOC} bumploc/
+ln -snf ${RDASApp}/fix/bumploc/${BUMPLOC} bumploc/${BUMPLOC}
```

Comment:
@SamuelDegelia-NOAA Thanks for working on this.

I am curious why I can run the current `link_mpasjedi_expr.sh` without a problem on Orion/Hercules at the command line?

Reply:
@guoqing-noaa Do you not get an error when running the link script via `build.sh`? Here is what happens for me with a fresh clone and without the fix in this PR: 

```
(base) orion-login-2[14] sdegelia$ ./build.sh
Building RDASApp on orion
  Build initiated Fri Sep  6 12:17:22 AM CDT 2024
-----------------------------------------------------------------------------------------------------------------------
The following dependent module(s) are not currently loaded: py-numpy/1.22.3 (required by: bufr/12.0.1), python/3.10.13 (required by: boost/1.83.0, bufr/12.0.1, fckit/0.11.0, atlas/0.35.1, py-pybind11/2.11.0)
-----------------------------------------------------------------------------------------------------------------------

Currently Loaded Modules:
  1) intel-oneapi-compilers/2023.1.0  17) netcdf-fortran/4.6.1  33) openblas/0.3.24   49) libxt/1.1.5
  2) stack-intel/2021.9.0             18) antlr/2.7.7           34) eckit/1.24.5      50) libxmu/1.1.4
  3) intel-oneapi-mpi/2021.9.0        19) gsl/2.7.1             35) fftw/3.3.10       51) libxpm/3.5.12
  4) stack-intel-oneapi-mpi/2021.9.0  20) nco/5.0.6             36) fckit/0.11.0      52) libxaw/1.0.13
  5) cmake/3.23.1                     21) parallelio/2.5.10     37) fiat/1.2.0        53) udunits/2.2.28
  6) nghttp2/1.57.0                   22) wget/1.21.1           38) ectrans/1.2.0     54) ncview/2.1.9
  7) curl/8.4.0                       23) pigz/2.7              39) qhull/2020.2      55) netcdf-cxx4/4.3.1
  8) zlib/1.2.13                      24) tar/1.34              40) atlas/0.35.1      56) py-pybind11/2.11.0
  9) git/2.31.1                       25) gettext/0.21.1        41) sp/2.5.0          57) contrib/0.1
 10) hdf5/1.14.0                      26) boost/1.83.0          42) gsl-lite/0.37.0   58) noaatools/3.1
 11) parallel-netcdf/1.12.2           27) bufr/12.0.1           43) libjpeg/2.1.0     59) rocoto/1.3.5
 12) snappy/1.1.10                    28) git-lfs/3.1.2         44) krb5/1.20.1       60) hpc/1.2.0
 13) zstd/1.5.2                       29) ecbuild/3.7.2         45) libtirpc/1.3.3    61) miniconda3/4.6.14
 14) c-blosc/1.21.5                   30) openjpeg/2.3.1        46) hdf/4.2.15        62) gdasapp/1.0.0
 15) netcdf-c/4.9.2                   31) eccodes/2.32.0        47) jedi-cmake/1.4.0  63) fms/2023.04
 16) nccmp/1.9.0.1                    32) eigen/3.4.0           48) libpng/1.6.37     64) RDAS/orion.intel



Linking in test data for MPAS-JEDI case
expdir is at: /work/noaa/wrfruc/sdegelia/RDASApp_dev3/RDASApp/expr/mpas_2024052700
ln: bumploc/: cannot overwrite directory
Configuring ...
```

After that the build continues successfully, but because of the line `ln: bumploc/: cannot overwrite directory`, none of the bumploc files get linked into `expr/mpas_2024052700/data/bumploc`.

Reply:
@SamuelDegelia-NOAA  I got the same error.  I was trying to understand why it was so. 

Finally, I recalled that global-workflow had a similar issue back in April.  
”Intermittent symlink failures on Rocky 9“:  https://github.com/NOAA-EMC/global-workflow/issues/2522
Per @WalterKolczynski-NOAA, it is "an issue deep in Lustre with Rocky 9".

I created a simple test.sh:
```
#!/bin/bash
rrfs-test/scripts/link_mpasjedi_expr.sh 
```
My testing revealed that `test.sh` always failed the first time and succeeded the second time.
and running `rrfs-test/scripts/link_mpasjedi_expr.sh` in the command line directly always succeeds.

Reply:
Ah, weird but interesting find! Thanks for the info.